packages/lib/src/smartthings-command.ts

```diff
@@ -145,70 +177,37 @@ export abstract class SmartThingsCommand extends Command implements SmartThingsC
 		return defaultValue
 	}
 
-	// eslint-disable-next-line @typescript-eslint/no-explicit-any
-	get args(): { [name: string]: any } {
-		if (!this._args) {
-			throw new Error('SmartThingsCommand not properly initialized')
-		}
-		return this._args
-	}
+	async init(): Promise<void> {
+		await super.init()
 
-	/**
-	 * Return input arguments, not including flags.
-	 */
-	get inputArgs(): string[] {
-		if (!this._argv) {
-			throw new Error('SmartThingsCommand not properly initialized')
-		}
-		return this._argv
-	}
+		this._logger = log4js.getLogger(`cli.${this.ctor.name}`)
 
-	// eslint-disable-next-line @typescript-eslint/no-explicit-any
-	get flags(): { [name: string]: any } {
-		if (!this._flags) {
-			throw new Error('SmartThingsCommand not properly initialized')
-		}
-		return this._flags
-	}
-
-	get profileName(): string {
-		if (!this._profileName) {
-			throw new Error('SmartThingsCommand not properly initialized')
-		}
-		return this._profileName
-	}
-
-	protected _tableGenerator?: TableGenerator
-	get tableGenerator(): TableGenerator {
-		if (!this._tableGenerator) {
-			throw new Error('SmartThingsCommand not properly initialized')
-		}
-		return this._tableGenerator
-	}
-
-	// eslint-disable-next-line @typescript-eslint/no-explicit-any
-	async setup(args: { [name: string]: any }, argv: string[], flags: { [name: string]: any }): Promise<void> {
+		const { args, argv, flags } = await this.parse(this.ctor)
 		this._args = args
 		this._argv = argv
 		this._flags = flags
 
-		this._profileName = flags.profile || 'default'
+		this._profileName = this.flags.profile || 'default'
 
-		this.cliConfig = await loadConfig({
+		this._cliConfig = await loadConfig({
 			configFilename: `${this.config.configDir}/config.yaml`,
 			managedConfigFilename: `${this.config.cacheDir}/config-managed.yaml`,
 			profileName: this.profileName,
 		})
+
 		this._profile = this.cliConfig.profile
 
 		const compact = this.flags.expanded
 			? false
 			: (this.flags.compact ? true : this.booleanConfigValue('compactTableOutput', true))
+
 		this._tableGenerator = new DefaultTableGenerator(compact)
 	}
 
 	/**
-	 * This method should be called when the user has decided to not complete a command.
+	 * @deprecated use Command.exit() instead
```

Comment:
Why not fix this method to use `Command.exit` instead? It also logs a message for the user which `Command.exit` doesn't do and is something we usually want to do in this case.